docs/howtos/DataPlane Generation - DPG/06types.mdx

```diff
@@ -5,9 +5,537 @@ import TabItem from "@theme/TabItem";
+  RED = "red"
+  BLUE = "blue"
+```
+
+</TabItem>
+<TabItem value="csharp" label="CSharp" >
+
+```csharp
+TODO;
+```
+
+</TabItem>
+<TabItem value="typescript" label="Typescript" >
+
+```typescript
+TODO;
+```
+
+</TabItem>
+<TabItem value="java" label="Java" >
+
+```java
+public final class Colors extends ExpandableStringEnum<Colors> {
+    public static final Colors RED = fromString("red");
+    public static final Colors BLUE = fromString("blue");
+}
+```
+
+</TabItem>
+</Tabs>
+
+### Unioned with Base Type Inline
+
+These are unions defined inline at point of usage which include the base type as an option.
+
+<Tabs>
+<TabItem value="typespec" label="TypeSpec" default>
+
+```typespec
+model Widget {
+  color: "red" | "blue" | string;
+}
+```
+
+</TabItem>
+<TabItem value="" label="Python">
+
+Python generates a union of literals again. Since we always generate our typing as open, this will
+look the same as our previous example where we generate an open union of strings.
+
+```python
+from typing import Literal
+
+type WidgetColor = Literal["red"] | Literal["blue"] | str
+
+model Widget:
+  color: WidgetColor
+```
+
+</TabItem>
+<TabItem value="csharp" label="CSharp" >
+
+```csharp
+TODO;
+```
+
+</TabItem>
+<TabItem value="typescript" label="Typescript" >
+
+```typescript
+TODO;
+```
+
+</TabItem>
+<TabItem value="java" label="Java" >
+
+```java
+public final class Color extends ExpandableStringEnum<Colors> {
+    public static final Color RED = fromString("red");
+    public static final Color BLUE = fromString("blue");
+}
+```
+
+</TabItem>
+</Tabs>
+
+### Extending unions
+
+For unions that include the shared base type of entries, each language will generate their version of an open enum.
+
+<Tabs>
+<TabItem value="typespec" label="TypeSpec" default>
+
+```typespec
+import "@azure-tools/typespec-azure-resource-manager";
+
+union ProvisioningState {
+  string,
+  inProgress: "inProgress",
+  Azure.ResourceManager.ResourceProvisioningState,
+}
+```
+
+</TabItem>
+<TabItem value="" label="Python">
+
+Python generates a single open enum.
+
+```python
+from enum import Enum
+from corehttp.utils import CaseInsensitiveEnumMeta
+
+class ProvisioningState(str, Enum, metaclass=CaseInsensitiveEnumMeta):
+  IN_PROGRESS = "inProgress"
+  SUCCEEDED = "Succeeded"
+  FAILED = "Failed"
+  CANCELED = "Canceled"
+```
+
+</TabItem>
+<TabItem value="csharp" label="CSharp" >
+
+```csharp
+TODO;
+```
+
+</TabItem>
+<TabItem value="typescript" label="Typescript" >
+
+```typescript
+TODO;
+```
+
+</TabItem>
+<TabItem value="java" label="Java" >
+
+```java
+public final class ProvisioningState extends ExpandableStringEnum<ProvisioningState> {
+    public static final ProvisioningState IN_PROGRESS = fromString("inProgress");
+    public static final ProvisioningState SUCCEEDED = fromString("Succeeded");
+    public static final ProvisioningState FAILED = fromString("Failed");
+    public static final ProvisioningState CANCELED = fromString("Canceled");
+}
```

Comment:
Behavior of union of union depends on `getUnionAsEnum` (I assume `ResourceProvisioningState` would be eventually a `union`).

However, in this case, if we do not flatten this, we still need a way to say `ProvisioningState` is something `inProgress` + those in `Azure.ResourceManager.ResourceProvisioningState`